### PR TITLE
Add `cairo_pie_output` flag to `cairo_vm_cli`

### PR DESCRIPTION
**Pull request by fmoletta**
This also involves:
- Fixing serialization of CairoPie to be fully compatible with the python version
- Implementing CairoPie::write_zip_file
- Adding comparison of cairo pie outputs between python and rust vm
Bonus:
- Moved handling of required and exclusive arguments to struct definition using clap derives

### CODE REVIEW

**Comment by Oppen** [comment]
Suggestion: maybe just write to a folder instead and use a tiny script for zipping?

File: vm/src/vm/runners/cairo_pie.rs [hunk 1]
```diff
@@ -199,6 +231,41 @@ mod serde_impl {
...
+            let mem_cap = values.len() * ADDR_BYTE_LEN + values.len() * FIELD_BYTE_LEN;
+            let mut res = Vec::with_capacity(mem_cap);
+
+            for ((segment, offset), value) in values.iter() {
+                let mem_addr = ADDR_BASE + *segment as u64 * OFFSET_BASE + *offset as u64;
+                res.extend_from_slice(mem_addr.to_le_bytes().as_ref());
+                match value {
+                    // Serializes RelocatableValue(little endian):
+                    // 1bit |   SEGMENT_BITS |   OFFSET_BITS
+                    // 1    |     segment    |   offset
```
Comment (Oppen) [inline vm/src/vm/runners/cairo_pie.rs:248]: This is 64 bits, so `u64` seems like a better choice than `BigUint`.

Comment (fmoletta) [reply vm/src/vm/runners/cairo_pie.rs:248]: Its 8 bits for the address (which is built as an u64 in line 243) but in this case we are handling a memory value that is a relocatable, which takes 32 bytes

**Review by Oppen — APPROVED** [review]

File: vm/src/vm/runners/cairo_pie.rs (hunk 1 above)
Comment (Oppen) [reply vm/src/vm/runners/cairo_pie.rs:248]: `Relocatable` is 8 bytes AFAIR.

**Review by Oppen — APPROVED** [review]

**Review by pefontana — APPROVED** [review]

**Pull request merged**